### PR TITLE
Improve messaging when timeshift token retrieval fails

### LINKED ISSUE #9813: Improve error message upon failing to retrieve multiplayer score submission token
**Describe the bug:** 
When trying to play multiplayer, the error notification appears and the map doesn't load.

Using arch linux with Xorg, and i3 as window managaer. Lazer compiled using '[pikaur](https://github.com/actionless/pikaur)' and the '[osu-lazer-git](https://aur.archlinux.org/packages/osu-lazer-git/)' package from AUR.

**Screenshots or videos showing encountered issue:**
https://youtu.be/t_1DFnhVumc

**osu!lazer version:**
Lazer 2020.806.0

**Logs:**
[database.log](https://github.com/ppy/osu/files/5045983/database.log)
[network.log](https://github.com/ppy/osu/files/5045984/network.log)
[performance.log](https://github.com/ppy/osu/files/5045985/performance.log)
[runtime.log](https://github.com/ppy/osu/files/5045986/runtime.log)

### PR DESCRIPTION
**Pull request by peppy**
Obviously not a final solution, but should better help self-compiling
(or unofficial package) users better understand why this is happening.

- Closes #9813

### CODE REVIEW

File: osu.Game/Screens/Multi/Play/TimeshiftPlayer.cs [hunk 1]
```diff
@@ -64,7 +64,7 @@ private void load()
             {
                 failed = true;
 
-                Logger.Error(e, "Failed to retrieve a score submission token.");
+                Logger.Error(e, "Failed to retrieve a score submission token.\n\nThis may happen if you are not running an official release of osu! (ie. you are self-compiling).");
```
Comment (bdach) [inline osu.Game/Screens/Multi/Play/TimeshiftPlayer.cs:67]: Wondering whether we want an android-specific message too as I imagine someone will inevitably roll along and ask why downloading the official APK causes this to pop up.

Comment (peppy) [reply osu.Game/Screens/Multi/Play/TimeshiftPlayer.cs:67]: android should work fine. there's already server-side allowance for this (see https://github.com/ppy/osu-web/pull/6449).

**Review by bdach — DISMISSED** [review]
works good :+1:

File: osu.Game/Screens/Multi/Play/TimeshiftPlayer.cs (hunk 1 above)
Comment (bdach) [reply osu.Game/Screens/Multi/Play/TimeshiftPlayer.cs:67]: Oh, right. Missed that as I don't look at the web channel that often.

Comment (smoogipoo) [inline osu.Game/Screens/Multi/Play/TimeshiftPlayer.cs:67]: This should probably also mention running an old version of osu!?

**Review by smoogipoo — CHANGES_REQUESTED** [review]

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**